src/header.rs

```diff
@@ -1,13 +1,97 @@
 use core::marker::PhantomData;
+use core::fmt::{Debug, Formatter};
+use core::cmp::Ordering;
 
-#[derive(Clone, Copy, Debug)]
+/// Possible Types of a [`Tag`]. The names and values are taken from the example C code
+/// at the bottom of the Multiboot2 specification.
+#[repr(u32)]
+#[derive(Copy, Clone, Debug)]
+pub enum TagType {
+    // 0
+    End,
+    Cmdline,
+    BootLoaderName,
+    Module,
+    BasicMeminfo,
+    // 5
+    Bootdev,
+    Mmap,
+    Vbe,
+    Framebuffer,
+    ElfSections,
+    // 10
+    Apm,
+    Efi32,
+    Efi64,
+    Smbios,
+    AcpiOld,
```

Comment:
As the maintainer of the `acpi` crates, I would probably prefer these to remain as `AcpiV1` and `AcpiV2` - I guess it can be argued that `V2` is confusing as it refers to later versions as well, but this would make it more consistent with the rest of the ecosystem.